csharp/src/Ice/IceSSL/SSLEngine.cs

```diff
@@ -0,0 +1,832 @@
+    {
+        if(_initialized)
+        {
+            return;
+        }
+
+        const string prefix = "IceSSL.";
+        Ice.Properties properties = communicator().getProperties();
+
+        //
+        // Check for a default directory. We look in this directory for
+        // files mentioned in the configuration.
+        //
+        _defaultDir = properties.getProperty(prefix + "DefaultDir");
+
+        string certStoreLocation = properties.getPropertyWithDefault(prefix + "CertStoreLocation", "CurrentUser");
+        StoreLocation storeLocation;
+        if(certStoreLocation == "CurrentUser")
+        {
+            storeLocation = StoreLocation.CurrentUser;
+        }
+        else if(certStoreLocation == "LocalMachine")
+        {
+            storeLocation = StoreLocation.LocalMachine;
+        }
+        else
+        {
+            _logger.warning(
+                "Invalid IceSSL.CertStoreLocation value `" + certStoreLocation + "' adjusted to `CurrentUser'");
+            storeLocation = StoreLocation.CurrentUser;
+        }
+        _useMachineContext = certStoreLocation == "LocalMachine";
+
+        //
+        // Protocols selects which protocols to enable, by default we only enable TLS1.0
+        // TLS1.1 and TLS1.2 to avoid security issues with SSLv3
+        //
+        var protocols = properties.getPropertyAsList(prefix + "Protocols");
+        if(protocols.Length > 0)
+        {
+            _protocols = parseProtocols(protocols);
+        }
+        else
+        {
+            _protocols = 0;
+            foreach(int v in Enum.GetValues(typeof(SslProtocols)))
+            {
+#pragma warning disable CS0618 // Type or member is obsolete
+                if (v > (int)SslProtocols.Ssl3 && v != (int)SslProtocols.Default)
+#pragma warning restore CS0618 // Type or member is obsolete
+                {
+                    _protocols |= (SslProtocols)v;
+                }
+            }
+        }
+        //
+        // CheckCertName determines whether we compare the name in a peer's
+        // certificate against its hostname.
+        //
+        _checkCertName = properties.getPropertyAsIntWithDefault(prefix + "CheckCertName", 0) > 0;
+
+        //
+        // VerifyDepthMax establishes the maximum length of a peer's certificate
+        // chain, including the peer's certificate. A value of 0 means there is
+        // no maximum.
+        //
+        _verifyDepthMax = properties.getPropertyAsIntWithDefault(prefix + "VerifyDepthMax", 3);
+
+        //
+        // CheckCRL determines whether the certificate revocation list is checked, and how strictly.
+        //
+        _checkCRL = properties.getPropertyAsIntWithDefault(prefix + "CheckCRL", 0);
+
+        //
+        // Check for a certificate verifier.
+        //
+        string certVerifierClass = properties.getProperty(prefix + "CertVerifier");
+        if(certVerifierClass.Length > 0)
+        {
+            // TODO Add a warning about this property?
```

Comment:
Is it worth having a warning for this? We removed the CertificateVerifier class, so probably don't need extra warning here.